File: tessoku/chapter5/a26.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	var q int
	fmt.Fscan(in, &q)

	for i := 0; i < q; i++ {
		var x int
		fmt.Fscan(in, &x)

		if isPrime(x) {
			fmt.Fprintln(out, "Yes")
		} else {
			fmt.Fprintln(out, "No")
		}
	}
}

func isPrime(num int) bool {
	if num < 2 {
		return false
	}

	if num == 2 {
		return true
	}

	top := math.Sqrt(float64(num))
	for i := 2; i <= int(top); i++ {
		if num%i == 0 {
			return false
		}
	}

	return true
}

File: tessoku/chapter5/a26.rb
def prime?(n)
  return false if n < 2
  return true if n == 2

  top = Math.sqrt(n).ceil
  (2..top).each do |i|
    return false if n % i == 0
  end

  true
end

q = gets.chomp.to_i
q.times do
  x = gets.chomp.to_i
  if prime?(x)
    puts "Yes"
  else
    puts "No"
  end
end
